### ISSUE TITLE
Implement Gemma 2 models

### ISSUE THREAD

**Issue by EricLBuehler** [EricLBuehler/mistral.rs#486]
Need to modify [Gemma model implementation](https://github.com/EricLBuehler/mistral.rs/blob/master/mistralrs-core/src/models/gemma.rs) with:

## Changelist over original Gemma and status:

- [x] Sliding window attn - **for layers that satisfy idx % 2 != 0, so every other, will use sliding window**
  - Affects KV cache retrieval
  - Affects sliding window mask generation 
- [x] Logit soft capping
  - In attention, between Q*K^T * s and matmul(V)
  ```py
   if self.config.attn_logit_softcapping is not None:
            attn_weights = attn_weights / self.config.attn_logit_softcapping
            attn_weights = torch.tanh(attn_weights)
            attn_weights = attn_weights * self.config.attn_logit_softcapping
  ``` 
  - After lm head
  ```py
  if self.config.final_logit_softcapping is not None:
            logits = logits / self.config.final_logit_softcapping
            logits = torch.tanh(logits)
            logits = logits * self.config.final_logit_softcapping
    ```
- [x] Use `query_pre_attn_scalar` instead of `1/sqrt(head_dim)`

## Links
- https://huggingface.co/google/gemma-2-9b-it
- https://github.com/huggingface/transformers/blob/main/src/transformers/models/gemma2/modeling_gemma2.py

**Comment by EricLBuehler** [comment]
Implemented in #490.
